### PR TITLE
Fix pretty printing and ReverseDiff constructor

### LINKED ISSUE #65: Broken constructor in v1.5
`ADTypes.AutoReverseDiff(false)` used to be a valid way to call the constructor before v1.5 and https://github.com/SciML/ADTypes.jl/commit/efdc12e72b44dba4f858bbef85c2835b2347f50e, but now only the keyword argument version works. I appreciate that the positional argument constructor was undocumented, but maybe a deprecation warning could still be put in place?

### PR DESCRIPTION
**Pull request by gdalle**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

Fix #65 and #66 

Should we test the round-trip-ability of these printings by parsing the string back into an object?

### CODE REVIEW

**Comment by gdalle** [comment]
**Code**

```julia
import ADTypes

struct FakeSparsityDetector <: ADTypes.AbstractSparsityDetector end
struct FakeColoringAlgorithm <: ADTypes.AbstractColoringAlgorithm end

for backend in [
    # dense
    ADTypes.AutoChainRules(; ruleconfig = :rc),
    ADTypes.AutoDiffractor(),
    ADTypes.AutoEnzyme(),
    ADTypes.AutoEnzyme(mode = :forward),
    ADTypes.AutoFastDifferentiation(),
    ADTypes.AutoFiniteDiff(),
    ADTypes.AutoFiniteDiff(fdtype = :fd, fdjtype = :fdj, fdhtype = :fdh),
    ADTypes.AutoFiniteDifferences(; fdm = :fdm),
    ADTypes.AutoForwardDiff(),
    ADTypes.AutoForwardDiff(chunksize = 3, tag = :tag),
    ADTypes.AutoPolyesterForwardDiff(),
    ADTypes.AutoPolyesterForwardDiff(chunksize = 3, tag = :tag),
    ADTypes.AutoReverseDiff(),
    ADTypes.AutoReverseDiff(compile = true),
    ADTypes.AutoSymbolics(),
    ADTypes.AutoTapir(),
    ADTypes.AutoTapir(safe_mode = false),
    ADTypes.AutoTracker(),
    ADTypes.AutoZygote(),
    # sparse
    ADTypes.AutoSparse(ADTypes.AutoForwardDiff()),
    ADTypes.AutoSparse(
        ADTypes.AutoForwardDiff();
        sparsity_detector = FakeSparsityDetector(),
        coloring_algorithm = FakeColoringAlgorithm()
    )
]
    println(backend)
end
```

**Before (1.5.1)**

```julia

```

**After (this PR, 1.5.2)**

```julia
ADTypes.AutoChainRules(ruleconfig=:rc)
ADTypes.AutoDiffractor()
ADTypes.AutoEnzyme()
ADTypes.AutoEnzyme(mode=:forward)
ADTypes.AutoFastDifferentiation()
ADTypes.AutoFiniteDiff()
ADTypes.AutoFiniteDiff(fdtype=:fd, fdjtype=:fdj, fdhtype=:fdh)
ADTypes.AutoFiniteDifferences(fdm=:fdm)
ADTypes.AutoForwardDiff()
ADTypes.AutoForwardDiff(chunksize=3, tag=:tag)
ADTypes.AutoPolyesterForwardDiff()
ADTypes.AutoPolyesterForwardDiff(chunksize=3, tag=:tag)
ADTypes.AutoReverseDiff()
ADTypes.AutoReverseDiff(compile=true)
ADTypes.AutoSymbolics()
ADTypes.AutoReverseDiff()
ADTypes.AutoReverseDiff(safe_mode=false)
ADTypes.AutoTracker()
ADTypes.AutoZygote()
ADTypes.AutoSparse(dense_ad=ADTypes.AutoForwardDiff())
ADTypes.AutoSparse(dense_ad=ADTypes.AutoForwardDiff(), sparsity_detector=FakeSparsityDetector(), coloring_algorithm=FakeColoringAlgorithm())

File: src/dense.jl [hunk 1]
```diff
@@ -111,21 +109,14 @@ end
...
-    end
-    s *= ")"
-    print(io, s)
+    print(io, AutoFiniteDiff, "(")
+    backend.fdtype != Val(:forward) &&
+        print(io, "fdtype=", repr(backend.fdtype; context = io), ", ")
+    backend.fdjtype != backend.fdtype &&
+        print(io, "fdjtype=", repr(backend.fdjtype; context = io), ", ")
+    backend.fdhtype != Val(:hcentral) &&
+        print(io, "fdhtype=", repr(backend.fdhtype; context = io))
```
Comment (ChrisRackauckas) [inline src/dense.jl:118]: These don't show the Val?

Comment (gdalle) [reply src/dense.jl:118]: Yes they do, there's just no `Val` in the test case I displayed above. Take a look here:

```julia
julia> string(AutoFiniteDiff(fdjtype=Val(:whatever)))
"AutoFiniteDiff(fdjtype=Val{:whatever}(), )"
```

Comment (gdalle) [reply src/dense.jl:118]: @ChrisRackauckas I think this one is a strict improvement over the current state, what do you say?

Comment (Vaibhavdixit02) [reply src/dense.jl:118]: LGTM

**Review by Vaibhavdixit02 — APPROVED** [review]

**Pull request merged**